File: Skills/goal-decomposition/scripts/prepare_goal_files.py
#!/usr/bin/env python3
import argparse
import json
import re
from datetime import datetime
from pathlib import Path


def main():
    parser = argparse.ArgumentParser()
    parser.add_argument("--project-root", required=True)
    args = parser.parse_args()

    root = Path(args.project_root).resolve()
    docs_dir = root / "docs" / "goal-decomposition"
    docs_dir.mkdir(parents=True, exist_ok=True)

    version_re = re.compile(r"_version_(\d+)\.md$", re.IGNORECASE)
    max_version = 0
    for p in docs_dir.glob("*_version_*.md"):
        m = version_re.search(p.name)
        if m:
            max_version = max(max_version, int(m.group(1)))

    version = max_version + 1
    stamp = datetime.now().strftime("%Y%m%d_%H%M%S")
    file_name = f"{stamp}_version_{version}.md"
    plan_file = docs_dir / file_name

    out = {
        "project_root": str(root),
        "docs_dir": str(docs_dir),
        "version": version,
        "timestamp": datetime.now().strftime("%Y-%m-%d %H:%M:%S"),
        "plan_file": str(plan_file),
    }
    print(json.dumps(out, ensure_ascii=False))


if __name__ == "__main__":
    main()

File: Skills/goal-decomposition/scripts/prepare_goal_files.js
#!/usr/bin/env bun
import path from "node:path"
import { mkdirSync, readdirSync } from "node:fs"

function arg(name) {
  const idx = process.argv.indexOf(name)
  return idx >= 0 ? process.argv[idx + 1] || "" : ""
}

const projectRootRaw = arg("--project-root")
if (!projectRootRaw) throw new Error("--project-root is required")
const root = path.resolve(projectRootRaw)
const docsDir = path.resolve(root, "docs", "goal-decomposition")
mkdirSync(docsDir, { recursive: true })
const version =
  readdirSync(docsDir)
    .map((name) => name.match(/_version_(\d+)\.md$/i))
    .filter(Boolean)
    .map((match) => Number(match[1]))
    .reduce((max, n) => (n > max ? n : max), 0) + 1
const now = new Date()
const pad = (n) => String(n).padStart(2, "0")
const stamp = `${now.getFullYear()}${pad(now.getMonth() + 1)}${pad(now.getDate())}_${pad(now.getHours())}${pad(now.getMinutes())}${pad(now.getSeconds())}`
const planFile = path.resolve(docsDir, `${stamp}_version_${version}.md`)
const iso = `${now.getFullYear()}-${pad(now.getMonth() + 1)}-${pad(now.getDate())} ${pad(now.getHours())}:${pad(now.getMinutes())}:${pad(now.getSeconds())}`
console.log(
  JSON.stringify({
    project_root: root,
    docs_dir: docsDir,
    version,
    timestamp: iso,
    plan_file: planFile,
  }),
)
